### PR TITLE
Add math test

### PR DESCRIPTION
**Pull request by nikolai-b**
After https://github.com/rails/arel/pull/449 was merged math can be done on these
nodes, adding a spec.

Copied over from https://github.com/rails/arel/pull/524

### CODE REVIEW

**Comment by kamipo** [comment]
Since Arel is private API, can we test this behavior from the public API?

**Comment by matthewd** [comment]
I think we can follow existing precedent for Arel tests. While it is private API, it exists to provide a clean layer to build AR on top of, so IMO it makes sense for it to have thorough unit tests.

**Comment by kamipo** [comment]
I see. So can we more exercise another expressions too (`maximum`, `minimum`, and `average`)?

**Comment by nikolai-b** [comment]
They are all defined 
https://github.com/rails/rails/blob/989b1cb4a326632a686d61df42695b27e4ef6b2e/activerecord/lib/arel/nodes/function.rb#L34-L40
and as far as I can see they don't have any tests: https://github.com/rails/rails/tree/master/activerecord/test/cases/arel/nodes .

Would you like me to add some?

**Comment by kamipo** [comment]
For now few tests exists in the file:

https://github.com/rails/rails/blob/2929d165c23f0d3976425a8e70de77847cc4b872/activerecord/lib/arel/expressions.rb#L4-L23

How about adding `math_test.rb` instead of adding one `.sum - 2` test case?
Not only `sum` and `count` but also attributes (e.g. `table[:id]`) can be mathmatically operated, but not tested.

**Comment by vedant1811** [comment]
Why is this still open?

**Comment by nikolai-b** [comment]
@kamipo I've added tests for the math operators.  Is that what you were suggesting?

File: activerecord/test/cases/arel/attributes/math_test.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        it "table node should be compatiable with #{math_operator}" do
+          table = Arel::Table.new :users
+          (table[:id].public_send(math_operator, 2)).to_sql.must_be_like %{
+            "users"."id" #{math_operator} 2
+          }
+        end
+      end
+
+      %i[+ - & | ^ << >>].each do |math_operator|
```
Comment (kamipo) [inline activerecord/test/cases/arel/attributes/math_test.rb]: Is there any reason not to include `*` and `/` in the list?

File: activerecord/test/cases/arel/attributes/math_test.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        end
+
+        it "minimum should be compatiable with #{math_operator}" do
+          table = Arel::Table.new :users
+          (table[:id].minimum.public_send(math_operator, 2)).to_sql.must_be_like %{
+            MIN("users"."id") #{math_operator} 2
+          }
+        end
+
+        it "table node should be compatiable with #{math_operator}" do
```
Comment (kamipo) [inline activerecord/test/cases/arel/attributes/math_test.rb]: `table[:id]` is not table node, but attribute node.

https://github.com/rails/rails/blob/df850b8d7126044a820291bbfd2bce80636a442c/activerecord/lib/arel/table.rb#L81-L83

https://github.com/rails/rails/blob/df850b8d7126044a820291bbfd2bce80636a442c/activerecord/lib/arel/attributes/attribute.rb#L5-L10

File: activerecord/test/cases/arel/attributes/math_test.rb (hunk 1 above)
Comment (nikolai-b) [reply activerecord/test/cases/arel/attributes/math_test.rb]: Yes, they have a slightly different SQL signature https://github.com/rails/rails/pull/32724/files/2bf08c892b1c14d5c4c433e791017dd8d499b21e#diff-e79446cf78f4dc85660b8a7c9bbec429R8

E.g.
```ruby
(AVG("users"."id") + 2)
```
vs
```ruby
AVG("users"."id") * 2
```

File: activerecord/test/cases/arel/attributes/math_test.rb (hunk 2 above)
Comment (nikolai-b) [reply activerecord/test/cases/arel/attributes/math_test.rb]: Thank you, I've updated it.

**Review by kamipo — APPROVED** [review]
Looks good to me. Can you squash your commits into one?

**Pull request merged**